src/org/exoplatform/controller/document/DocumentAdapter.java

```diff
@@ -67,123 +68,165 @@ public long getItemId(int pos) {
         return pos;
     }
 
-    @Override
-    public View getView(int position, View convertView, ViewGroup parent) {
-
-        final int pos = position;
-        LayoutInflater inflater = (LayoutInflater) _mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
-        // View rowView = inflater.inflate(R.layout.fileitem, parent, false);
-
-        final ExoFile myFile = _documentList.get(pos);
-
-        if ("".equals(myFile.name) && "".equals(myFile.path)) {
-            convertView = inflater.inflate(R.layout.gadget_tab_layout, parent, false);
-            TextView textViewTabTitle = (TextView) convertView.findViewById(R.id.textView_Tab_Title);
-            if (myFile.driveName.equals(ExoConstants.DOCUMENT_PERSONAL_DRIVER))
-                textViewTabTitle.setText(_mContext.getResources().getString(R.string.Personal));
-            else if (myFile.driveName.equals(ExoConstants.DOCUMENT_GROUP_DRIVER))
-                textViewTabTitle.setText(_mContext.getResources().getString(R.string.Group));
-            else if (myFile.driveName.equals(ExoConstants.DOCUMENT_GENERAL_DRIVER))
-                textViewTabTitle.setText(_mContext.getResources().getString(R.string.General));
-            return (convertView);
-        } else {
-            convertView = inflater.inflate(R.layout.fileitem, parent, false);
-
-            Button btnAction = (Button) convertView.findViewById(R.id.Button_FileAction);
-            ImageView icon = (ImageView) convertView.findViewById(R.id.icon);
-            TextView lb = (TextView) convertView.findViewById(R.id.label);
-            // using the natural name here is possible
-            lb.setText(myFile.getName());
-
-            final ExoFile file = DocumentActivity._documentActivityInstance._fileForCurrentActionBar;
-            if ("".equals(myFile.currentFolder)) {
-                /*
-                 * If current folder is null, make the action button is
-                 * invisible
-                 */
-                btnAction.setVisibility(View.INVISIBLE);
+  private static final int VIEW_TYPE_GADGET   = 0;
```

Comment:
There's no gadget thing in the documents. You should rename this into `VIEW_TYPE_DRIVE_ITEM` as these views represent ECMS drives. Rename other variables and classes accordingly, i.e. GadgetHolder -> DriveHolder, etc.